[Ng Jian Yi] iP

ngjianyi opened the pull request:
# Dudu
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

- text-based
- VERY easy to learn
- ~FAST~  SUPER FAST to use

All you need is to do is,

1. download is from [here](https://github.com/ngjianyi/ip/releases/tag/v0.1)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :laughing:

and it is **FREE**

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders

Code snippet of our `main` method
```java
public static void main(String[] args) {
    new Dudu("./data/dudu.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ngjianyi committed:
Add greeting

ngjianyi committed:
Rename main class

ngjianyi committed:
Add support for echo

ngjianyi committed:
Add support for list

ngjianyi committed:
Added support for marking tasks

ngjianyi committed:
Moved classes to inidivdual files

ngjianyi committed:
Removed nested classes from main

ngjianyi committed:
Edited code to use String.format

ngjianyi committed:
Created ToDo class

ngjianyi committed:
Added Deadline class

ngjianyi committed:
Added Event class

ngjianyi committed:
Added checking of tasks in main

ngjianyi committed:
Added tests in input and expected files

ngjianyi committed:
Improved regex

ngjianyi committed:
Added MissingDescriptionException class

ngjianyi committed:
Added missing description error for todo

ngjianyi committed:
Added getDetails function

ngjianyi committed:
Created MissingDateTimeException class

ngjianyi committed:
Added missing date time exception

ngjianyi committed:
Modified error checking for event

ngjianyi committed:
Modified error checking for deadline

ngjianyi committed:
Added error checking for invalid count for mark and unmark

ngjianyi committed:
Modified getDetails function

ngjianyi committed:
Changed array to ArrayList

ngjianyi committed:
Implemented delete function

ngjianyi committed:
Created InvalidFormatException class

ngjianyi committed:
Added enum and changed to switch case

ngjianyi committed:
Added break to while loop in main

ngjianyi committed:
Add support for creating directory and files

ngjianyi committed:
Add support for todo task from text file

ngjianyi committed:
Add preliminary support for deadline and event tasks

ngjianyi committed:
Fix support for deadline and event tasks

ngjianyi committed:
Add formatString method to tasks

ngjianyi committed:
Add filewriter to addTask function

ngjianyi committed:
Fixed filewriter in addTask function

ngjianyi committed:
Add rewriteFile function to mark, unmark and delete tasks

ngjianyi committed:
Remove unused file imports

ngjianyi committed:
Merge branch 'branch-Level-7'
Add local storage functionality

ngjianyi committed:
Change Deadline class to store LocalDate object

ngjianyi committed:
Change Event class to store LocalDate object

ngjianyi committed:
Change main to use LocalDate

ngjianyi committed:
Change formatString DateTimeFormatter format

ngjianyi committed:
Merge branch 'branch-Level-8'
Change date to be stored in LocalDate object

ngjianyi committed:
Create Storage class

ngjianyi committed:
Add error handling to Storage load function

ngjianyi committed:
Edit creation of folders and file in Storage constructor

ngjianyi committed:
Use Storage class in Dudu main

ngjianyi committed:
Move TaskType enum from Dudu to Storage

ngjianyi committed:
Add rewriteFile method in Storage class

ngjianyi committed:
Add TaskList class

ngjianyi committed:
Add UI class

ngjianyi committed:
Add abstract Command class

ngjianyi committed:
Add Parser file

ngjianyi committed:
Add Command classes

ngjianyi committed:
Remove unnecessary code from Dudu

ngjianyi committed:
Merge branch 'MoreOOP'
Refactor code to use OOP

ngjianyi committed:
Move classes into packages

ngjianyi committed:
Merge branch 'Packages'
Move classes into packages

ngjianyi committed:
Add gradle support

ngjianyi committed:
Add gradle

ngjianyi committed:
Add JUnit test

ngjianyi committed:
Add find command

ngjianyi committed:
Edit addTask method in TaskList

ngjianyi committed:
Add JavaDocs for Command classes

ngjianyi committed:
Add JavaDocs for exceptions and task

ngjianyi committed:
Add find command

ngjianyi committed:
Fix merge conflict

ngjianyi committed:
Change coding standard for tasks

ngjianyi committed:
Fixed merge conflict for A-CodingStandard branch

ngjianyi committed:
Add checkstyle files

ngjianyi committed:
Edit import order for test files

ngjianyi committed:
Improve style of files

ngjianyi committed:
Merge branch 'branch-A-CheckStyle'

nathanshew commented on src/main/java/dudu/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+     *
+     * @return true if the command will cause the application to exit, false otherwise.
+     */
+    public abstract boolean isExit();
```
I like that your boolean sounds like a boolean!

nathanshew commented on src/main/java/dudu/command/CommandUnmark.java:
```diff
@@ -0,0 +1,50 @@
+
+import dudu.task.Task;
+import dudu.utils.Storage;
+import dudu.utils.TaskList;
```
I like that you listed all your imported classes explicitly!

nathanshew commented on src/main/java/dudu/command/CommandMark.java:
```diff
@@ -0,0 +1,50 @@
+
+import java.io.IOException;
+
+import dudu.task.Task;
+import dudu.utils.Storage;
+import dudu.utils.TaskList;
+import dudu.utils.UI;
```
I like that the ordering of import statements are consistent.

nathanshew commented on src/main/java/dudu/command/CommandEvent.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public class CommandEvent extends Command {
+    private Task task;
+
```
I like how all your indentations are 4 spaces

nathanshew commented on src/main/java/dudu/utils/Storage.java:
```diff
@@ -0,0 +1,143 @@
+    public ArrayList<Task> load() {
+        File file = new File(filePath);
+        file.getParentFile().mkdirs();
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                String line = sc.nextLine();
+                try {
+                    addTask(line, tasks);
+                } catch (InvalidFormatException e) {
+                    System.out.println(e);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println(e);
+        }
```
I like how your try-catch statements have the correct form!

nathanshew reviewed (commented):
Well done!

ngjianyi committed:
Add files for javafx gui

ngjianyi committed:
Modify execute methods in Command classes

ngjianyi committed:
Edit JavaDocs of UI class

ngjianyi committed:
Add styling to chatbot

ngjianyi committed:
Add css styling

ngjianyi committed:
Remove line wrap around output messages

ngjianyi committed:
Merge branch 'branch-Level-10'

ngjianyi committed:
Add varargs to find command

ngjianyi committed:
Merge branch 'branch-A-Varargs'

Timenikhil commented on .gitignore:
```diff
@@ -15,3 +15,5 @@ bin/
 /text-ui-test/ACTUAL.TXT
 text-ui-test/EXPECTED-UNIX.TXT
+
+/data/
```
I like that you have added /data to .gitignore

Timenikhil commented on src/main/java/dudu/Dudu.java:
```diff
@@ -0,0 +1,53 @@
+     *
+     * @return Welcome message
+     */
+    public String welcomeUser() {
```
your function naming is quite descriptive and concise

Timenikhil commented on src/main/java/dudu/command/CommandBye.java:
```diff
@@ -0,0 +1,32 @@
+
+/**
+ * Represents a bye user command into the chatbot
+ */
```
does this really need to abstracted out into its own class?

Timenikhil commented on src/main/java/dudu/command/CommandDeadline.java:
```diff
@@ -0,0 +1,62 @@
+/**
+ * Represents a deadline task creation user command into the chatbot
+ */
+public class CommandDeadline extends Command {
```
Could this have been named better? Would DeadlineCommand be more well ordered?

Timenikhil commented on src/main/java/dudu/gui/Launcher.java:
```diff
@@ -0,0 +1,12 @@
+/**
+ * A launcher class to workaround classpath issues.
+ */
+public class Launcher {
```
Could the class name be made more descriptive?

Timenikhil commented on src/main/java/dudu/task/Task.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Represents an abstract class for a task
+ */
+public class Task {
```
I like that you have used a noun to show the Task class

Timenikhil commented on src/main/java/dudu/utils/Storage.java:
```diff
@@ -0,0 +1,143 @@
+     * @param taskList The TaskList instance containing the updated list of tasks
+     * @throws IOException If there is an error during saving the task to storage.
+     */
+    public void rewriteFile(TaskList taskList) throws IOException {
```
I like that you have used an action for the file rewrite action

Timenikhil commented on src/main/java/dudu/utils/UI.java:
```diff
@@ -0,0 +1,129 @@
+     *
+     * @return Help message
+     */
+    public String helpMessage() {
```
Could this be changed to be a verb instead of a noun?

Timenikhil commented on src/main/java/dudu/utils/Parser.java:
```diff
@@ -0,0 +1,134 @@
+
+    public static int getIndex(String[] input) throws MissingDescriptionException {
+        if (input.length <= 1 || input[1].replaceAll("\\D+", "").isEmpty()) {
+            throw new MissingDescriptionException("Please input a number");
```
I like your descriptive exception names

Timenikhil reviewed (commented):
Looks good to me. Quite a lot of classes though. Could consider compactifying.

ngjianyi committed:
Command classes: add assert to execute methods

Command classes do not have error checking for execute methods.

Assert is essential to check for unexpected errors during runtime.

Assert is added within the execute method to check for errors from TaskList class.

Assert provides a simple and lightweight way to detect errors during runtime.

ngjianyi committed:
Edit bye, help and greeting message functions

ngjianyi committed:
Parser class: Edit getContent method

getContent method checks for missing content by splitting the user input with a space, before checking
for the length of the array and indexing the array at the second element to check the string.

getContent method does not check for missing content correctly. It incorrectly checks the split command
by array length and empty content in the same if condition.

Let's separate the two conditions into two separate if conditions instead of putting it together.

By separating the two checks, it accurately checks the array length first before trying to index the array.

ngjianyi committed:
Refactor parser methods to reduce length of functions

ngjianyi committed:
Refactor Storage methods

ngjianyi committed:
TaskList class: Use streams for findTask method

ngjianyi committed:
Merge pull request #1 from ngjianyi/branch-A-Assertions

Command classes: add assert to execute methods

ngjianyi committed:
Merge master to Code Quality branch

ngjianyi committed:
Merge branch 'master' into branch-A-Streams

ngjianyi committed:
Merge pull request #2 from ngjianyi/branch-A-CodeQuality

Merge Code Quality branch with master

ngjianyi committed:
Merge branch 'master' into branch-A-Streams

ngjianyi committed:
Merge pull request #3 from ngjianyi/branch-A-Streams

Merge Streams branch with master

ngjianyi committed:
Add undo command

ngjianyi committed:
Merge branch 'Undo'

ngjianyi committed:
Improve task classes code quality

ngjianyi committed:
Edit execute method for command classes

ngjianyi committed:
Add InvalidCommand class

ngjianyi committed:
Add AddCommand class and remove individual task commands

ngjianyi committed:
Edit command class names

ngjianyi committed:
Rename command classes

ngjianyi committed:
Edit UI get message methods

ngjianyi committed:
Edit JavaDocs for command classes

ngjianyi committed:
Edit JavaDocs

ngjianyi committed:
Edit JUnit tests

ngjianyi committed:
Add Ui.png

ngjianyi committed:
Add user guide to README

ngjianyi committed:
Add commands to titles in README

ngjianyi committed:
Merge branch 'branch-A-UserGuide'

ngjianyi committed:
Fix delete on empty list bug

ngjianyi committed:
Change error message in Parser

ngjianyi committed:
Command classes: change to use DuduException instead of IOException

ngjianyi committed:
Remove unused imports

ngjianyi committed:
Fix delete 0 bug

ngjianyi committed:
Add system exit to bye command
